pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -704,7 +692,37 @@ func (vs *VSphere) InstanceExistsByProviderID(ctx context.Context, providerID st
 
 // InstanceShutdownByProviderID returns true if the instance is in safe state to detach volumes
 func (vs *VSphere) InstanceShutdownByProviderID(ctx context.Context, providerID string) (bool, error) {
-	return false, cloudprovider.NotImplemented
+	nodeName, err := vs.GetNodeNameFromProviderID(providerID)
+	if err != nil {
+		glog.Errorf("Error while getting nodename for providerID %s", providerID)
+		return false, err
+	}
+
+	// Create context
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
You can use the `ctx` param that was passed to `InstanceShutdownByProviderID` as-is, no need for cancel here.